docs/runs.md

```diff
@@ -64,6 +66,62 @@ A `Run` definition supports the following fields:
 
 ### Specifying the target Custom Task
 
+A custom task resource's `Spec` may be directly embedded in the `Run` or it may
+be referred to by a `Ref`. But, not both at the same time.
+
+1. [Specifying the target Custom Task by embedding it's spec](#specifying-the-target-custom-task-by-embedding-its-spec)
+2. [Specifying the target Custom Task with ref](#specifying-the-target-custom-task-with-ref)
+
+### Specifying the target Custom Task by embedding it's spec
+To specify the custom task spec, it can be embedded directly into a
+`Run`'s spec as shown below:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  name: embedded-run
+spec:
+  spec:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+    spec:
+      field1: value1
+      field2: value2
+```
+
+This initiates the execution of a `Run` of a custom task of type `Example`, in
+the `example.dev` API group,  with the version `v1alpha1`.
+
+#### Developer guide for custom controllers supporting `spec`.
+
+1. A custom controller may support either of `Ref` or `Spec` or both. In case, it does not
+   support one of them, custom controller should respond with proper validation error.
+   
+2. It is highly recommended that the custom controller has appropriate validations
+   to handle `Run` which misses a `Ref` or a `Spec`.
+
+3. In case of an embedded `spec`, a custom task controller may not actually create
```

Comment:
I'm not sure we need this?  It's up to the custom controller to decide what they need.

In Tekton we never create a resource for embedded specs - it does not add any value - or worst, it creates confusion, since it's not clear anymore which version of the resource is being executed, the one on the cluster or the one embedded.

So if we say something I think it should be that we discourage custom controllers from storing the embedded custom resource in `etcd`.

Reply:
You are right, it can be confusing and it is up to the custom controller, which way they want to implement.